library/ssl_misc.h

```diff
@@ -1743,4 +1751,57 @@ static inline const void *mbedtls_ssl_get_groups( const mbedtls_ssl_context *ssl
     #endif
 }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+/* Corresponding PSA algorithm for MBEDTLS_CIPHER_NULL.
+ * Same value is used fo PSA_ALG_CATEGORY_CIPHER, hence it is
+ * guaranteed to not be a valid PSA algorithm identifier.
+ */
+#define MBEDTLS_SSL_NULL_CIPHER 0x04000000
+
+/**
+ * \brief       Translate mbedtls cipher type/taglen pair to psa:
+ *              algorithm, key type and key size.
+ *
+ * \param  mbedtls_cipher_type [in] given mbedtls cipher type
+ * \param  taglen              [in] given tag length
+ *                                  0 - default tag length
+ * \param  alg                 [out] corresponding PSA alg
+ *                                   There is no corresponding PSA
+ *                                   alg for MBEDTLS_SSL_NULL_CIPHER, so
+ *                                   MBEDTLS_SSL_NULL_CIPHER is returned
+ * \param  key_type            [out] corresponding PSA key type
+ * \param  key_size            [out] corresponding PSA key size
+ *
+ * \return                     PSA_SUCCESS on success or PSA_ERROR_NOT_SUPPORTED if
+ *                             conversion is not supported.
+ */
+psa_status_t mbedtls_ssl_cipher_to_psa( mbedtls_cipher_type_t mbedtls_cipher_type,
+                                    size_t taglen,
+                                    psa_algorithm_t *alg,
+                                    psa_key_type_t *key_type,
+                                    size_t *key_size );
+
+/**
+ * \brief       Convert given PSA status to mbedtls error code.
+ *
+ * \param  status      [in] given PSA status
+ *
+ * \return             corresponding mbedtls error code
+ */
+static inline int psa_status_to_mbedtls( psa_status_t status )
+{
+    switch( status )
+    {
+        case PSA_SUCCESS:
+            return( 0 );
+        case PSA_ERROR_INSUFFICIENT_MEMORY:
+            return( MBEDTLS_ERR_CIPHER_ALLOC_FAILED );
+        case PSA_ERROR_NOT_SUPPORTED:
+            return( MBEDTLS_ERR_CIPHER_FEATURE_UNAVAILABLE );
+        default:
```

Comment:
Actually we also want to convert `PSA_ERROR_INVALID_SIGNATURE` to `MBEDTLS_ERR_SSL_INVALID_MAC` because that value is handled specially in the rest of the library (especially for DTLS where we often want to ignore a few unauthentic, but not too many of them) and in tests.

This is the direct of indirect cause of the following tests failures:
```
DTLS proxy: inject invalid AD record, badmac_limit 1;pattern 'too many records with bad MAC' MUST be present in the Server output
DTLS proxy: inject invalid AD record, badmac_limit 2, exchanges 2;pattern 'discarding invalid record (mac)' MUST be present in the Client output
DTLS proxy: inject invalid AD record, badmac_limit 2;bad client exit code (expected 0, got 1)
DTLS proxy: inject invalid AD record, default badmac_limit;bad client exit code (expected 0, got 1)
DTLS wrong PSK: badmac alert;pattern 'SSL - Verification of the message MAC failed' MUST be present in the Server output
ECJPAKE: password mismatch, DTLS;pattern 'SSL - Verification of the message MAC failed' MUST be present in the Server output
ECJPAKE: password mismatch, TLS;pattern 'SSL - Verification of the message MAC failed' MUST be present in the Server output
```